forge-game/src/main/java/forge/game/card/CardZoneTable.java

```diff
@@ -63,11 +61,19 @@ public CardCollectionView getLastStateGraveyard() {
         return lastStateGraveyard;
     }
     public void setLastStateBattlefield(CardCollectionView lastState) {
-        // store it in a new object, it might be from Game which can also refresh itself
-        this.lastStateBattlefield = new CardCollection(lastState);
+        if (lastState == null) {
+            lastStateBattlefield = CardCollection.EMPTY;
+        } else {
+            // store it in a new object, it might be from Game which can also refresh itself
+            lastStateBattlefield = new CardCollection(lastState);
```

Comment:
should this be short-style?

`lastStateBattlefield = lastState == null ? CardCollection.EMPTY : new CardCollection(lastState);`